Scripts/Services/Revamped Dungeons/WrongDungeon/Mobile/Fezzik.cs

```diff
@@ -50,54 +55,43 @@ public Fezzik(Serial serial)
 
         public override int Meat { get { return 2; } }
 
-        public void SpawnGreenGoo(Mobile target)
+        public override void AlterDamageScalarFrom(Mobile caster, ref double scalar)
         {
-            Map map = this.Map;
+            if (0.5 >= Utility.RandomDouble())
+                this.SpawnGreenGoo();
+        }
 
-            if (map == null)
-                return;
+        public override void OnGotMeleeAttack(Mobile attacker)
+        {
+            base.OnGotMeleeAttack(attacker);
 
-            int newGreenGoo = Utility.RandomMinMax(3, 6);
+            if (0.5 >= Utility.RandomDouble())
+                this.SpawnGreenGoo();
+        }
 
-            for (int i = 0; i < newGreenGoo; ++i)
+        public void SpawnGreenGoo()
+        {
+            if (this.m_StinkingCauldronTime <= DateTime.UtcNow)
             {
-                GooeyMaggots goo = new GooeyMaggots();
-
-                goo.Team = this.Team;
-                goo.FightMode = FightMode.Closest;
-
-                bool validLocation = false;
-                Point3D loc = this.Location;
+                new StinkingCauldron().MoveToWorld(this.Location, this.Map);
 
-                for (int j = 0; !validLocation && j < 10; ++j)
-                {
-                    int x = this.X + Utility.Random(3) - 1;
-                    int y = this.Y + Utility.Random(3) - 1;
-                    int z = map.GetAverageZ(x, y);
-
-                    if (validLocation = map.CanFit(x, y, this.Z, 16, false, false))
-                        loc = new Point3D(x, y, this.Z);
-                    else if (validLocation = map.CanFit(x, y, z, 16, false, false))
-                        loc = new Point3D(x, y, z);
-                }
-
-                goo.MoveToWorld(loc, map);
-                goo.Combatant = target;
+                this.m_StinkingCauldronTime = DateTime.UtcNow + TimeSpan.FromMinutes(2);
             }
         }
 
-        public override void AlterDamageScalarFrom(Mobile caster, ref double scalar)
+        public override void OnDeath(Container c)
         {
-            if (0.5 >= Utility.RandomDouble())
-                this.SpawnGreenGoo(caster);
+            base.OnDeath(c);
+
+            if (0.2 > Utility.RandomDouble())
+            {
+                c.DropItem(new RecipeScroll(603));
```

Comment:
Avoid magic numbers, you can replace this by `(int) CookRecipes.ThreeTierCakee` to improve readability and prevent bugs if changing the `CookRecipes` enumeration.